prow/cmd/mkbuild-cluster/main.go

```diff
@@ -0,0 +1,331 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"encoding/base64"
+	"errors"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/ghodss/yaml"
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/test-infra/prow/kube"
+)
+
+const (
+	useClientCertEnv = "CLOUDSDK_CONTAINER_USE_CLIENT_CERTIFICATE"
+)
+
+var (
+	coder = base64.StdEncoding
+)
+
+type options struct {
+	account       string
+	alias         string
+	changeContext bool
+	cluster       string
+	getClientCert bool
+	overwrite     bool
+	printEntry    bool
+	printData     bool
+	project       string
+	skipCheck     bool
+	zone          string
+}
+
+type describe struct {
+	Auth     describeAuth `json:"masterAuth"`
+	Endpoint string       `json:"endpoint"`
+}
+
+type describeAuth struct {
+	ClientCertificate    string `json:"clientCertificate"`
+	ClientKey            string `json:"clientKey"`
+	ClusterCACertificate string `json:"clusterCaCertificate"`
+}
+
+func parseOptions() options {
+	var o options
+	if err := o.parseArgs(flag.CommandLine, os.Args[1:]); err != nil {
+		logrus.Fatalf("Invalid flags: %v", err)
+	}
+	return o
+}
+
+func (o *options) parseArgs(flags *flag.FlagSet, args []string) error {
+	flags.StringVar(&o.account, "account", "", "use this account to describe --cluster")
+	flags.StringVar(&o.alias, "alias", "", "the --build-cluster alias to add")
+	flags.StringVar(&o.cluster, "cluster", "", "the GKE cluster to describe")
+	flags.StringVar(&o.project, "project", "", "the GKE project to describe")
+	flags.StringVar(&o.zone, "zone", "", "the GKE zone to describe")
+	flags.BoolVar(&o.printData, "print-file", false, "print the file outside of the configmap secret")
+	flags.BoolVar(&o.printEntry, "print-entry", false, "print the new entry without appending to existing ones at stdin")
+	flags.BoolVar(&o.getClientCert, "get-client-cert", false, fmt.Sprintf("first get-credentials for the cluster using %s=True", useClientCertEnv))
+	flags.BoolVar(&o.changeContext, "change-context", false, "allow --get-client-cert to change kubectl config current-context")
+	flags.BoolVar(&o.skipCheck, "skip-check", false, "skip validating the creds work in a client")
+	switch err := flags.Parse(args); {
+	case err != nil:
+		return err
+	case o.cluster == "":
+		return errors.New("--cluster required")
+	case o.project == "":
+		return errors.New("--project required")
+	case o.zone == "":
+		return errors.New("--zone required")
+	case o.alias == "":
+		return fmt.Errorf("--alias required (use %q for default)", kube.DefaultClusterAlias)
+	}
+	return nil
+}
+
+func main() {
+	// Gather options from flags
+	o := parseOptions()
+	if err := do(o); err != nil {
+		logrus.Fatalf("Failed: %v", err)
+	}
+}
+
+// useContext calls kubectl config use-context ctx
+func useContext(o options, ctx string) error {
+	_, cmd := command("kubectl", "config", "use-context", ctx)
```

Comment:
Why not set up the client library appropriately?

Reply:
Ack